SC3001 with busybox shell: "In dash, process substitution is not supported."

#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC3001
- My shellcheck version (`shellcheck --version` or "online"): 0.11.0 (Debian Sid package 0.11.0-1)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# shellcheck shell=busybox
while read -r n; do
    echo "$n"
done < <(find /home)
```

#### Here's what shellcheck currently says:

```
Line 5:
done < <(find /home)
       ^-- SC3001 (error): In dash, process substitution is not supported.
```

#### Here's what I wanted or expected to see:

No error. The code works fine in Busybox' ash, and the shell type is not dash either way.

--- a/src/ShellCheck/Checks/ShellSupport.hs
+++ b/src/ShellCheck/Checks/ShellSupport.hs
@@ -245,6 +245,7 @@ prop_checkBashisms147 = verify checkBashisms "[ -R ref ]" -- SC3063
 prop_checkBashisms148 = verify checkBashisms "[ -N file ]" -- SC3064
 prop_checkBashisms149 = verify checkBashisms "[ -G file ]" -- SC3066
 prop_checkBashisms150 = verify checkBashisms "[ -O file ]" -- SC3067
+prop_checkBashisms151 = verifyNot checkBashisms "#!/bin/busybox sh\npwd | tee >(gzip > log.gz)" -- SC3001
 checkBashisms = ForShell [Sh, Dash, BusyboxSh] $ \t -> do
     params <- ask
     kludge params t
@@ -260,7 +261,7 @@ checkBashisms = ForShell [Sh, Dash, BusyboxSh] $ \t -> do
         else warn id code $ "In POSIX sh, " ++ s ++ " undefined."
     asStr = getLiteralString
 
-    bashism (T_ProcSub id _ _) = warnMsg id 3001 "process substitution is"
+    bashism (T_ProcSub id _ _) = unless isBusyboxSh $ warnMsg id 3001 "process substitution is"
     bashism (T_Extglob id _ _) = warnMsg id 3002 "extglob is"
     bashism (T_DollarDoubleQuoted id _) = warnMsg id 3004 "$\"..\" is"
     bashism (T_ForArithmetic id _ _ _ _) = warnMsg id 3005 "arithmetic for loops are"

